### PR TITLE
[SPARK-20300][ML][PYSPARK] Python API for ALSModel.recommendForAllUsers,Items

### PR DESCRIPTION
**Pull request by MLnick**
Add Python API for `ALSModel` methods `recommendForAllUsers`, `recommendForAllItems`

## How was this patch tested?

New doc tests.

### CODE REVIEW

**Comment by MLnick** [comment]
cc @jkbradley

File: python/pyspark/ml/recommendation.py [hunk 1]
```diff
@@ -384,6 +392,28 @@ def itemFactors(self):
...
+                 stored as an array of (itemCol, rating) Rows.
+        """
+        return self._call_java("recommendForAllUsers", numItems)
+
+    @since("2.2.0")
+    def recommendForAllItems(self, numUsers):
+        """
+        Returns top `numUsers` users recommended for each item, for all items.
+
+        :param numItems: max number of recommendations for each item
```
Comment (viirya) [inline python/pyspark/ml/recommendation.py]: numItems -> numUsers

**Comment by viirya** [comment]
LGTM except for a doc comment.

File: python/pyspark/ml/recommendation.py (hunk 1 above)
Comment (MLnick) [inline python/pyspark/ml/recommendation.py]: Thanks!

**Comment by MLnick** [comment]
If no other comments (@jkbradley) I will merge to branch-2.2 in a few days.

**Comment by MLnick** [comment]
Merged to master/branch-2.2

**Pull request closed without merging**